test/particle.test.ts

```diff
@@ -55,4 +56,29 @@ describe("Can create a particle", () => {
         expect(p.info()).toEqual(
             [{dim: [1], length: 1, name: "x"}]);
     })
+
+    it("Can copy state into a vector view", () => {
+        const n = 5;
+        const state = dustState(n, 3);
+        const pars = {n, sd: 1};
+        const p = new Particle(new models.Walk(base, pars), 0);
+        p.setState([2, 3, 4, 5, 6]);
+        // Copy the full state into the second particle
+        p.copyState(null, state.viewParticle(1));
+        expect(state.getParticle(0)).toEqual([0, 0, 0, 0, 0]);
+        expect(state.getParticle(1)).toEqual([2, 3, 4, 5, 6]);
+    });
+
+    it("Can copy partial into a vector view", () => {
+        const n = 5;
+        const index = [1, 3];
+        const state = dustState(index.length, 3);
+        const pars = {n, sd: 1};
+        const p = new Particle(new models.Walk(base, pars), 0);
```

Comment:
I'm a bit confused about what this state copying functionality is used for. So you can copy a subset of state from a particle into a particle with smaller state, but you can't copy a subset of state between two same sized particles? 

I.e. you can't do 
```
const state = dustState(n, 3);
p.copyState(index, state.viewParticle(1));
expect(state.getParticle(1)).toEqual([0, 3, 0, 5, 0]);
```

Reply:
yeah, copying between particles will be done particle-to-particle rather than via a state.

The state objects are for reporting back during a model run - so we create a `DustStateTime` object that covers a simulation, then call this `copyState` method to fill in the state. This means that we can assemble the big 3d array of outputs without having to create a triple-nested array, and we can easily do indexing against that later to slice out data for plotting and summarising